### PR TITLE
Add (basic) Civo Instance Support

### PR DESCRIPTION
**Pull request by dirien**
### Description of your changes
Add (basic) Civo Instance Support

Fixes #

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by AlexsJones** [comment]
Thank you so much for your contribution @dirien 
We will review it shortly and update this issue.

File: pkg/civocli/cli.go [hunk 1]
```diff
@@ -12,12 +13,42 @@ var (
...
 
+func emptyIfNil(in *string) string {
+	if in == nil {
+		return ""
+	}
+	return *in
+}
+
+// GenerateObservation creates the CivoInstanceObservation from instance infos
+func GenerateObservation(instance *civogo.Instance) (v1alpha1.CivoInstanceObservation, error) {
```
Comment (AlexsJones) [inline pkg/civocli/cli.go:37]:
```suggestion
func GenerateInstanceObservation(instance *civogo.Instance) (v1alpha1.CivoInstanceObservation, error) {
```

File: internal/controller/civoinstance/civoinstance.go [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+	}
+	civoInstance, err := e.civoClient.GetInstance(cr.Status.AtProvider.ID)
+	if err != nil {
+		return managed.ExternalObservation{ResourceExists: false}, err
+	}
+	if civoInstance == nil {
+		return managed.ExternalObservation{ResourceExists: false}, nil
+	}
+
+	cr.Status.AtProvider, err = civocli.GenerateObservation(civoInstance)
```
Comment (AlexsJones) [inline internal/controller/civoinstance/civoinstance.go:127]:
```suggestion
cr.Status.AtProvider, err = civocli.GenerateInstanceObservation(civoInstance)
```

**Comment by AnaisUrlichs** [comment]
Looks great, thank you so much for adding instance support 🥳

File: pkg/civocli/cli.go (hunk 1 above)
Comment (dirien) [reply pkg/civocli/cli.go:37]: Hi @AlexsJones, sorry i was on vacation and did not get this mail :(

File: internal/controller/civoinstance/civoinstance.go (hunk 2 above)
Comment (dirien) [reply internal/controller/civoinstance/civoinstance.go:127]: Hi @AlexsJones, sorry i was on vacation and did not get this mail :(

**Comment by dirien** [comment]
hi @AnaisUrlichs, thanks for the feedback! So happy i can be part of this provider!

**Pull request merged**